tests/src/Container/ConstructorTest.php

```diff
@@ -73,4 +73,19 @@ public function testScalar()
         // assert
         $this->assertInternalType('int', $entry->getNumber());
     }
+
+    public function testScalarWithClass()
+    {
+        $container = new Container([
+            Foo::class => [],
+            ScalarDependency::class => [
+                'args' => [Foo::class]
+            ],
+        ]);
+        // act
+        /** @var ScalarDependency $entry */
+        $entry = $container->get(ScalarDependency::class);
+        // assert
+        $this->assertInternalType('string', $entry->getNumber());
```

Comment:
I guess for this case using `assertEquals` is more suitable.

Reply:
also, I think method `getNumber` needs rename to `getString`?

Reply:
In my point of view, this method should have name `getValue` instead of typed name

Reply:
👍